### PR TITLE
Change regularization in SR

### PR DESCRIPTION
**Pull request by camelto2**
Please review the [developer documentation](https://github.com/QMCPACK/qmcpack/wiki/Development-workflow)
on the wiki of this project that contains help and requirements.

## Proposed changes
This PR changes the regularization scheme back to just adding a constant to the diagonal of the overlap matrix. If S becomes ill-conditioned, the CG can be slower to converge. In #5156, I changed the regularization to what was supposed to be a scaling of the diagonal elements which was argued in a paper from Sorella to be a better regularization. Effectively, in matrix notation, this would be S-> S(I+diag(S)*eps). Unfortunately, I implemented S(I+eps * I) which scales all elements of S, not just the diagonal. That operation obviously does not reduce the condition number, and some simple test problems I was playing around with were running slower since the CG step was taking longer. 

Also, doing some simple numerical experiments on diagonally dominant matrices similar to the overlap matrix, it seems that adding a constant to the diagonal has a larger reduction in the condition number than the scaling the diagonal for a fixed value of the scaling constant epsilon. So for now, it seems better to just revert back to how I implemented it in the original PR. 

I also updated the documentation and changed the name of the method as Ye requested in #5017 after it was merged. 

## What type(s) of changes does this code introduce?
_Delete the items that do not apply_

- Bugfix

### Does this introduce a breaking change?

- No

## What systems has this change been tested on?
mac 

## Checklist

_Update the following with a yes where the items apply. If you're unsure about any of them, don't hesitate to ask.  This is
simply a reminder of what we are going to look for before merging your code._

- Yes. This PR is up to date with current the current state of 'develop'
- Yes. Code added or changed in the PR has been clang-formatted
- No. This PR adds tests to cover any new code, or to catch a bug that is being fixed
- Yes. Documentation has been added (if appropriate)

### CODE REVIEW

File: docs/methods.rst [hunk 1]
```diff
@@ -1174,7 +1174,7 @@ Additional information and recommendations:
    parameters discussed earlier for descent are useful for setting up
    the descent engine to do this averaging on its own.
 
-Stochastic Reconfiguration
+Stochastic Reconfiguration with Gradient Descent
```
Comment (ye-luo) [inline docs/methods.rst:1177]: Can we have `Conjugate Gradient` instead of `Gradient Descent`?

Comment (camelto2) [reply docs/methods.rst:1177]: yep thats what I meant to put. Working too fast. Will fix

File: docs/methods.rst [hunk 2]
```diff
@@ -1174,7 +1174,7 @@ Additional information and recommendations:
    parameters discussed earlier for descent are useful for setting up
    the descent engine to do this averaging on its own.
 
-Stochastic Reconfiguration
+Stochastic Reconfiguration with Conjugate Gradient
 ^^^^^^^^^^^^^^^^^^^^^^^^^^
```
Comment (ye-luo) [inline docs/methods.rst:1178]: also need to add enough `^^^` otherwise we get warnings when compiling the docs.

Comment (camelto2) [reply docs/methods.rst]: again moving too fast

**Review by ye-luo — APPROVED** [review]

**Comment by ye-luo** [comment]
Test this please

**Pull request merged**